### PR TITLE
give YAML_CPP_INSTALL a default value that can be overridden

### PR DESCRIPTION
**Pull request by jasonbeach**
simple cmake fix that adds a default value of OFF when yaml-cpp is not the master project, but allows a user to override that.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -93,7 +99,8 @@ endif()
 
 target_compile_options(yaml-cpp
   PRIVATE
-    $<${not-msvc}:-Wall -Wextra -Wshadow -Weffc++ -Wno-long-long>
+    $<${not-msvc}:-Wall -Wextra -Wshadow -Wno-long-long>
```
Comment (jbeder) [inline CMakeLists.txt:102]: This looks separate from the primary point of the PR?

File: CMakeLists.txt [hunk 2]
```diff
@@ -17,16 +17,22 @@ include(CTest)
 
 find_program(YAML_CPP_CLANG_FORMAT_EXE NAMES clang-format)
 
+if (NOT DEFINED YAML_CPP_MASTER_PROJECT)
```
Comment (jbeder) [inline CMakeLists.txt:20]: Is this pattern common for CMake? It seems like you're trying to solve a pretty common problem.

**Review by jbeder — CHANGES_REQUESTED** [review]
Can you clarify in the PR description what problem this is solving?

**Comment by jasonbeach** [comment]
> Can you clarify in the PR description what problem this is solving?

By making creation of the install targets a cmake dependent option, there's no way to override it.  This makes using it with FetchContent not possible when the consuming library wants to be installed. Issue #1003 and #794 (duplicates of each other).  By making it an option the way I have, a consuming library can override it and force the creation of the install targets.

I'm very curious on why a number of very well known libraries such as this one, libfmt, spdlog, etc) follow the pattern of not automatically creating the install targets when they're not the top level project.  I guess I don't understand the use case or when that would be useful. Could you shed some light on that?

File: CMakeLists.txt (hunk 1 above)
Comment (jasonbeach) [reply CMakeLists.txt:102]: it is, but adding the `YAML_CPP_MASTER_PROJECT` option from this PR makes this fix very easy.  It's a partial fix for issue #970 (basically if you're not the toplevel project omit the -Weffc++ option.)  If you'd rather once the other change is approve I can make a new PR for this.

Comment (jasonbeach) [reply CMakeLists.txt:102]: I guess I need to add the `not-msvc` to the check as well (the option would be erroneously added on msvc when then top-level project) ...may be just as easy to omit all these options if this library is not the top-level project.

File: CMakeLists.txt (hunk 2 above)
Comment (jasonbeach) [reply CMakeLists.txt:20]: I'm very curious on why a number of very well known libraries such as this one, libfmt, spdlog, etc) follow the pattern of not automatically creating the install targets when they're not the top level project.  I guess I don't understand the use case or when that would be useful. Could you shed some light on that?

Comment (jasonbeach) [reply CMakeLists.txt:20]: The big reason the fix is needed is that the current implementation breaks the install ability in projects that include this via CMake's FetchContent. (CMake complains target xxx requires yaml-cpp which is not going to be installed (or something to that effect)

**Comment by jbeder** [comment]
Obviated by #1007.

**Pull request closed without merging**